static-instrumenter/agent-instrumenter/src/integrationTest/java/JarTest.java

```diff
@@ -43,17 +46,25 @@ void testSampleJar() throws Exception {
             outPath.toString());
 
     Process instrumentationProcess = instrumentationProcessBuilder.start();
-    InputStream instrumentationIn = instrumentationProcess.getErrorStream();
     instrumentationProcess.waitFor(10, TimeUnit.SECONDS);
 
-    String instrumentationLog =
-        new String(instrumentationIn.readAllBytes(), StandardCharsets.UTF_8);
+    boolean isHotJdkJvm = System.getProperty("java.vm.name").contains("OpenJDK");
+    if (isHotJdkJvm) {
+      InputStream errorOutput = instrumentationProcess.getErrorStream();
+      String errorOutputAsString = new String(errorOutput.readAllBytes(), StandardCharsets.UTF_8);
+      assertThat(errorOutputAsString)
+          .isNotEmpty()
+          .contains(
+              "Sharing is only supported for boot loader classes because bootstrap classpath has been appended");
+    }
+
     assertThat(instrumentationProcess.exitValue()).isEqualTo(0);
-    assertThat(instrumentationLog).isNotEmpty();
 
     Path resultAppPath = outPath.resolve("app.jar");
+    assertThat(resultAppPath).exists();
     assertThat(Files.exists(resultAppPath)).isTrue();
```

Comment:
These two lines assert the same thing

Reply:
I have forgotten to remove the previous way of doing this assertion. Fixed.